bundles/org.openhab.binding.siemensrds/src/main/java/org/openhab/binding/siemensrds/internal/RdsAccessToken.java

```diff
@@ -58,71 +68,100 @@
      * private method: execute the HTTP POST on the server
      */
     private static String httpGetTokenJson(String apiKey, String user, String password) {
```

Comment:
This method is very big and the exception handling feels strange, I would prefer to have it split up in smaller method with each having there own responsibility. 

About the exception handling:
- I doubt that an empty String is valid return for this method, would it make more sense to rethrow a specific exception to indicate that you failed, most likely you can not continue with the empty token
- The way that exceptions work is intended to allow you to separate exception handling and the actual logic, so you could consider to have bigger blocks of code with a single try-catch, now you have 5 separate try-catch blocks in one method
- Why not just put logging statement for the impossible condition?

Reply:
Hi @martinvw, we definitely need feedback from @Hilbrand on this. Reason is that I had originally written this method with one large try/catch block, but @Hilbrand asked me to use try-with-resources instead, so I rewrote it. But only some of the objects used support try-with-resources, and some of them do not support it, so I was forced to use several smaller try/catch or try-with-resources blocklets. Of course I can write it any way you prefer, but I need to have a common agreement between the reviewers how you do actually prefer it.

PS I do indeed return an empty JSON string in case of error, and in the calling code, I check isEmpty() on the string before processing further, or not. Personally I think an isEmpty() check (or a != null check) is as good a way of checking a valid return, as creating and throwing a custom exception. But again it is I think a matter of taste, and I need to know what your common agreed taste is. :)

Note that all these errors are sometimes to be expected, but not regular exceptions, which would occur only from time to time, as when for example the Siemens cloud server would be down for maintenance. It is just necessary for the binding to handle those kind of things softly..

Reply:
I think it's not needed to put into separate methods. But it would be better to still have the try-catch around the code. I intended the use of try-with-resources not to specifically also handle the catch, but let the catching be handled by the large try-catch.

I don't think there is a specific preference for throwing exceptions or returning empty data. It's up to the binding developer I guess. One guideline I use: Most error cases might be of interest to the user. For example by setting the thing offline with a useful message. By returning an empty string it's probably more difficult to report to the user what the cause of the problem was and in such case the user needs to look into the log what is exactly wrong. So I would prefer throwing a (custom) exception. But I don't think this is something really enforce.

Reply:
Ok, I pushed a new version of the code, in which I have 1) used try-with-resources without catch blocks, and therefore 2) centralised the catch blocks into a single outer try block. Furthermore I have moved the exception catching an extra level up from the HTTP level into the higher level calling methods, so that I can centralise the catching of HTTP errors and JSON errors into a single catch block, which just contains a single Log.warn() message that shows the originating exception's cause and message.

Please note that these changes have not actually changed the functionality at all; they are just playing around with different coding styles. So I do hope we can move on fairly fast with merging this PR again. :)